docs/workflows/kserve-predictor.md

```diff
@@ -0,0 +1,24 @@
+# KServe predictor extension for FuseML workflows
+
+## Overview
+
+The KServe predictor workflow step can be used to create and manage [KServe prediction services](https://kserve.github.io/website/) to serve input ML models as part of the execution of FuseML workflows. The KServe predictor is designed to work primarily with the following types of ML models that are trained and saved using the MLflow library:
+
+- scikit-learn pickled models
+- TensorFlow (saved_model) models
+- ONNX models
+
+The KServe predictor step expects a model URL to be supplied as input, pointing to the location in an MLflow artifact store where the model is stored. Currently, S3 is the only protocol supported for the MLflow artifact store back-end.
+
+The predictor performs the following tasks:
+
+- downloads the model locally from the MLflow artifact store
+- if so instructed, it auto-detects the model format based on the information stored in the MLflow artifact store and decides which KServe predictor engine to use for it. Otherwise, it validates the model format against the type of predictor engine specified as input.
```

Comment:
Maybe you can write explicitly the rules for such selections

Reply:
Totally agree, but I'll have to do it in a follow-up PR. This is just a draft I had to include in this PR to silence the CI.